src/workerd/server/tests/weakref/test.mjs

```diff
@@ -0,0 +1,48 @@
+import { env } from 'node:process';
+import { beforeEach, afterEach, test } from 'node:test';
+import assert from 'node:assert';
+import { WorkerdServerHarness } from '../server-harness.mjs';
+
+// Global that is reset for each test.
+let workerd;
+
+assert(
+  env.WORKERD_BINARY !== undefined,
+  'You must set the WORKERD_BINARY environment variable.'
+);
+assert(
+  env.WORKERD_CONFIG !== undefined,
+  'You must set the WORKERD_CONFIG environment variable.'
+);
+
+// Start workerd.
+beforeEach(async () => {
+  workerd = new WorkerdServerHarness({
+    workerdBinary: env.WORKERD_BINARY,
+    workerdConfig: env.WORKERD_CONFIG,
+
+    // Hard-coded to match a socket name expected in the `workerdConfig` file.
+    listenPortNames: ['http'],
+  });
+
+  await workerd.start();
+
+  await workerd.getListenPort('http');
+});
+
+// Stop workerd.
+afterEach(async () => {
+  const [code, signal] = await workerd.stop();
+  assert(code === 0 || signal === 'SIGTERM');
```

Comment:
would prefer if this were somehow reworked to use `assert.strictEqual(...)` to ensure a more useful stack output if the assertion fails